### PR TITLE
Test other schemas

### PR DESCRIPTION
**Pull request by tschmidtb51**
- prepare test scripts to work with any example uploaded
- improve scripts
- add test script to test validate/data
- add to the workflow

### CODE REVIEW

File: csaf_2.0/test/aggregator_schema/run_tests.sh [hunk 1]
```diff
@@ -23,16 +24,22 @@ validate() {
 
 }
 
+
+test_all() {
+  for i in `ls -1 $TESTPATH`
```
Comment (sthagen) [inline csaf_2.0/test/aggregator_schema/run_tests.sh:29]: Maybe `for i in $(ls -1 "${TESTPATH}")`?

File: csaf_2.0/test/aggregator_schema/run_tests.sh [hunk 2]
```diff
@@ -23,16 +24,22 @@ validate() {
...
+    validate $i
+  done
+} 
+
 SCHEMA=$ORIG_SCHEMA
-validate csaf_2.0/examples/aggregator/example-01-aggregator.json
-validate csaf_2.0/examples/aggregator/example-02-aggregator.json
+test_all
  
 echo -n "Generating strict schema ... "
```
Comment (sthagen) [inline csaf_2.0/test/aggregator_schema/run_tests.sh:38]: Maybe `printf "%s" "Generating strict schema ... "`?

File: csaf_2.0/test/aggregator_schema/run_tests.sh [hunk 3]
```diff
@@ -23,16 +24,22 @@ validate() {
...
 python3 $STRICT_GENERATOR $ORIG_SCHEMA > $AGGREGATOR_STRICT_SCHEMA
```
Comment (sthagen) [inline csaf_2.0/test/aggregator_schema/run_tests.sh:39]: Maybe `python3 "${STRICT_GENERATOR}" "${ORIG_SCHEMA}" > "${AGGREGATOR_STRICT_SCHEMA}"` ?

File: csaf_2.0/test/aggregator_schema/run_tests.sh [hunk 4]
```diff
@@ -23,16 +24,22 @@ validate() {
...
 echo done
```
Comment (sthagen) [inline csaf_2.0/test/aggregator_schema/run_tests.sh:40]: Maybe `printf "%s\n" "done"`?

**Review by sthagen — APPROVED** [review]

File: csaf_2.0/test/validator/run_tests.sh [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+    printf "%s\n" SUCCESS
+  else
+    printf "%s\n" FAILED
+    FAIL=1
+  fi
+
+}
+
+test_all() {
+  for i in `ls -1 ${TESTPATH} | grep -v $EXCLUDE`
```
Comment (tschmidtb51) [inline csaf_2.0/test/validator/run_tests.sh:27]: @sthagen Any idea for this one? I wasn't able to remove the backticks...

**Review by tschmidtb51 — COMMENTED** [review]
@sthagen Anyy other suggestions to improve the quality?

File: csaf_2.0/test/validator/run_tests.sh (hunk 5 above)
Comment (sthagen) [reply csaf_2.0/test/validator/run_tests.sh:27]: `$(ls -1 ${TESTPATH} | grep -v $EXCLUDE)` should do.

**Review by sthagen — APPROVED** [review]

File: csaf_2.0/test/validator/run_tests.sh (hunk 5 above)
Comment (tschmidtb51) [reply csaf_2.0/test/validator/run_tests.sh]: Done in [c3fb0f5](https://github.com/oasis-tcs/csaf/pull/439/commits/c3fb0f5735858723e901a544a88b7ac35b21c20d).

**Pull request merged**